pkg/gitops/flux/flux.go

```diff
@@ -53,35 +53,35 @@ const (
 	deleteClusterconfigCommitMessage  = "Delete commit of cluster configuration; generated by EKS-A CLI"
 )
 
-type FluxAddonClient struct {
-	flux      Flux
+type Flux struct {
+	flux      Client
 	gitTools  *gitFactory.GitTools
 	cliConfig *config.CliConfig
 	retrier   *retrier.Retrier
 }
 
-func NewFluxAddonClient(flux Flux, gitTools *gitFactory.GitTools, cliConfig *config.CliConfig) *FluxAddonClient {
-	return &FluxAddonClient{
+func NewFlux(flux Client, gitTools *gitFactory.GitTools, cliConfig *config.CliConfig) *Flux {
+	return &Flux{
 		flux:      flux,
 		gitTools:  gitTools,
 		cliConfig: cliConfig,
 		retrier:   retrier.NewWithMaxRetries(maxRetries, backOffPeriod),
 	}
 }
 
-// Flux is an interface that abstracts the basic commands of flux executable.
-type Flux interface {
-	// BootstrapToolkitsComponentsGithub bootstraps toolkit components in a GitHub repository.
-	BootstrapToolkitsComponentsGithub(ctx context.Context, cluster *types.Cluster, fluxConfig *v1alpha1.FluxConfig) error
+// Client is an interface that abstracts the basic commands of flux executable.
+type Client interface {
```

Comment:
nit: should we rename it to FluxClient?

Reply:
I think it's ok since out of this pkg we will call `flux.Client` instead of `flux.FluxClient` which is duplicate

Reply:
package name is Flux, so Client is the go-prefered name in this case